src/bin/cratesfyi.rs

```diff
@@ -325,3 +36,402 @@ fn logger_init() {
+
+    /// Skips building documentation if documentation exists
+    #[structopt(name = "SKIP_IF_EXISTS", short = "s", long = "skip")]
+    skip_if_exists: bool,
+
+    /// Skips building documentation if build log exists
+    #[structopt(name = "SKIP_IF_LOG_EXISTS", long = "skip-if-log-exists")]
+    skip_if_log_exists: bool,
+
+    /// Keeps build directory after build.
+    #[structopt(
+        name = "KEEP_BUILD_DIRECTORY",
+        short = "k",
+        long = "keep-build-directory"
+    )]
+    keep_build_directory: bool,
+
+    #[structopt(subcommand)]
+    subcommand: BuildSubcommand,
+}
+
+impl Build {
+    pub fn handle_args(self) {
+        let docbuilder = {
+            let mut doc_options = DocBuilderOptions::from_prefix(self.prefix);
+
+            if let Some(crates_io_index_path) = self.crates_io_index_path {
+                doc_options.crates_io_index_path = crates_io_index_path;
+            }
+
+            doc_options.skip_if_exists = self.skip_if_exists;
+            doc_options.skip_if_log_exists = self.skip_if_log_exists;
+            doc_options.keep_build_directory = self.keep_build_directory;
+
+            doc_options
+                .check_paths()
+                .expect("The given paths were invalid");
+
+            DocBuilder::new(doc_options)
+        };
+
+        self.subcommand.handle_args(docbuilder);
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, StructOpt)]
+enum BuildSubcommand {
+    /// Builds documentation of every crate
+    World,
+
+    /// Builds documentation for a crate
+    Crate {
+        /// Crate name
+        #[structopt(
+            name = "CRATE_NAME",
+            required_unless("LOCAL"),
+            requires("CRATE_VERSION")
+        )]
+        crate_name: String,
+
+        /// Version of crate
+        #[structopt(name = "CRATE_VERSION")]
+        crate_version: String,
+
+        /// Build a crate at a specific path
+        #[structopt(short = "l", long = "local", conflicts_with_all(&["CRATE_NAME", "CRATE_VERSION"]))]
+        local: Option<PathBuf>,
+    },
+
+    /// update the currently installed rustup toolchain
+    UpdateToolchain {
+        /// Update the toolchain only if no toolchain is currently installed
+        #[structopt(name = "ONLY_FIRST_TIME", long = "only-first-time")]
+        only_first_time: bool,
+    },
+
+    /// Adds essential files for the installed version of rustc
+    AddEssentialFiles,
+
+    /// Locks cratesfyi daemon to stop building new crates
+    Lock,
+
+    /// Unlocks cratesfyi daemon to continue building new crates
+    Unlock,
+
+    PrintOptions,
+}
+
+impl BuildSubcommand {
+    pub fn handle_args(self, mut docbuilder: DocBuilder) {
+        match self {
+            Self::World => {
+                docbuilder.load_cache().expect("Failed to load cache");
+
+                let mut builder = RustwideBuilder::init().unwrap();
+                builder
+                    .build_world(&mut docbuilder)
```

Comment:
This command needs to be rethought in general :/ It takes several months to run.

Reply:
Yah, the only time I can ever see this used is for a self-hosted instance

Reply:
Let's keep it for now, I might have a use for it (for the staging server, applied on the staging index).